### PR TITLE
Stabilize VS Code extension E2E state assertions

### LINKED ISSUE #19547: [CI Failure] Flaky: VS Code extension E2E (Windows, edge-cases) shard fails with generic exit code, unrelated to PR changes
## Build Information

Build: https://github.com/microsoft/aspire/actions/runs/32403422458
Build error leg or test failing: Tests / Hosting.Azure / Hosting.Azure (ubuntu-latest) / `VS Code extension E2E (Windows, edge-cases)`
Pull request: #19238

## Error Message

```
Process completed with exit code 1. (Run extension E2E tests step failed; detailed job logs unavailable: 'Failed to fetch logs for job')
```

## Description

Flaky: VS Code extension E2E (Windows, edge-cases) shard fails with generic exit code, unrelated to PR changes

**Type**: flaky-test

## Occurrences

| Date | Build | Job | PR |
|------|-------|-----|----|
| 2026-08-20 | [32403422458](https://github.com/microsoft/aspire/actions/runs/32403422458) | Tests / Hosting.Azure / Hosting.Azure (ubuntu-latest) | #19238 |
| 2026-08-22 | [32543765057](https://github.com/microsoft/aspire/actions/runs/32543765057) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, debug-dashboard) | #19131 |
| 2026-08-22 | [32548071306](https://github.com/microsoft/aspire/actions/runs/32548071306) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19300 |
| 2026-08-22 | [32550757022](https://github.com/microsoft/aspire/actions/runs/32550757022) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19577 |
| 2026-08-22 | [32551216411](https://github.com/microsoft/aspire/actions/runs/32551216411) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19494 |
| 2026-08-22 | [32554012820](https://github.com/microsoft/aspire/actions/runs/32554012820) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19516 |
| 2026-08-24 | [32675340121](https://github.com/microsoft/aspire/actions/runs/32675340121) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, workspace-target-proof) | #19044 |
| 2026-08-24 | [32689591099](https://github.com/microsoft/aspire/actions/runs/32689591099) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19429 |
| 2026-08-24 | [32755490521](https://github.com/microsoft/aspire/actions/runs/32755490521) | Tests / Hosting.Valkey / Hosting.Valkey (ubuntu-latest) | #19613 |
| 2026-08-24 | [32756925219](https://github.com/microsoft/aspire/actions/runs/32756925219) | Tests / Acquisition / Acquisition (windows-latest) | #19592 |
| 2026-08-24 | [32758526447](https://github.com/microsoft/aspire/actions/runs/32758526447) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19590 |
| 2026-08-24 | [32774802376](https://github.com/microsoft/aspire/actions/runs/32774802376) | Tests / Cli / Cli (windows-latest) | #19048 |
| 2026-08-24 | [32781735084](https://github.com/microsoft/aspire/actions/runs/32781735084) | Tests / Infrastructure / Infrastructure (ubuntu-latest) | #19624 |
| 2026-08-25 | [32785638179](https://github.com/microsoft/aspire/actions/runs/32785638179) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19583 |
| 2026-08-25 | [32791546024](https://github.com/microsoft/aspire/actions/runs/32791546024) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19430 |
| 2026-08-25 | [32792740958](https://github.com/microsoft/aspire/actions/runs/32792740958) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19608 |
| 2026-08-25 | [32812190860](https://github.com/microsoft/aspire/actions/runs/32812190860) | Tests / Cli / Cli (windows-latest) | #19032 |
| 2026-08-25 | [32814771945](https://github.com/microsoft/aspire/actions/runs/32814771945) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #18924 |
| 2026-08-25 | [32834178883](https://github.com/microsoft/aspire/actions/runs/32834178883) | Tests / Hosting.Radius / Hosting.Radius (ubuntu-latest) | #19656 |
| 2026-08-25 | [32864200821](https://github.com/microsoft/aspire/actions/runs/32864200821) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19627 |
| 2026-08-25 | [32880916910](https://github.com/microsoft/aspire/actions/runs/32880916910) | Tests / Dashboard / Dashboard (windows-latest) | #19620 |
| 2026-08-25 | [32883688176](https://github.com/microsoft/aspire/actions/runs/32883688176) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19600 |
| 2026-08-25 | [32885558008](https://github.com/microsoft/aspire/actions/runs/32885558008) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, workspace-target-proof) | #19615 |
| 2026-08-25 | [32898142217](https://github.com/microsoft/aspire/actions/runs/32898142217) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19641 |
| 2026-08-26 | [32909638198](https://github.com/microsoft/aspire/actions/runs/32909638198) | Tests / Dashboard / Dashboard (ubuntu-latest) | #19677 |
| 2026-08-26 | [32917733900](https://github.com/microsoft/aspire/actions/runs/32917733900) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, edge-cases) | #19690 |
| 2026-08-26 | [32978808010](https://github.com/microsoft/aspire/actions/runs/32978808010) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, edge-cases) | #19698 |
| 2026-09-17 | [35275989131](https://github.com/microsoft/aspire/actions/runs/35275989131) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, edge-cases) | #20203 |

### LINKED ISSUE #19576: [CI Failure] Flaky: VS Code extension E2E (Windows, dynamic-debug-configuration) shard fails with generic exit code 1, unrelated to PR changes
## Build Information

Build: https://github.com/microsoft/aspire/actions/runs/32505042756
Build error leg: Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration)
Pull request: #19510

## Error Message

```
Process completed with exit code 1. (Run extension E2E tests step failed; job logs unavailable: 'Failed to fetch logs for job'). No TRX test failures extracted. Occurs on 'VS Code extension E2E (Windows, dynamic-debug-configuration)' shard.
```

## Description

Flaky: VS Code extension E2E (Windows, dynamic-debug-configuration) shard fails with generic exit code 1, unrelated to PR changes

**Type**: flaky-test

## Occurrences

| Date | Build | Job | PR |
|------|-------|-----|----|
| 2026-08-21 | [32505042756](https://github.com/microsoft/aspire/actions/runs/32505042756) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19510 |
| 2026-09-02 | [33583332995](https://github.com/microsoft/aspire/actions/runs/33583332995) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19762 |
| 2026-09-02 | [33654701260](https://github.com/microsoft/aspire/actions/runs/33654701260) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19820 |
| 2026-09-02 | [33662594149](https://github.com/microsoft/aspire/actions/runs/33662594149) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19865 |
| 2026-09-02 | [33669706128](https://github.com/microsoft/aspire/actions/runs/33669706128) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19872 |
| 2026-09-03 | [33710947040](https://github.com/microsoft/aspire/actions/runs/33710947040) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19807 |
| 2026-09-03 | [33789740828](https://github.com/microsoft/aspire/actions/runs/33789740828) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19868 |
| 2026-09-04 | [33879357897](https://github.com/microsoft/aspire/actions/runs/33879357897) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19929 |
| 2026-09-04 | [33887016795](https://github.com/microsoft/aspire/actions/runs/33887016795) | Tests / No-package tests (regular, Aspire.Hosting.Radius.Tests, Hosting.Radius, Hosting.Radius, tests/Asp... / Hosting.Radius (windows-latest) | #19478 |
| 2026-09-04 | [33910459971](https://github.com/microsoft/aspire/actions/runs/33910459971) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19197 |
| 2026-09-04 | [33911967287](https://github.com/microsoft/aspire/actions/runs/33911967287) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #18196 |
| 2026-09-08 | [34251319766](https://github.com/microsoft/aspire/actions/runs/34251319766) | Tests / No-package tests / Hosting-4 (windows-latest) | #19678 |
| 2026-09-08 | [34268996559](https://github.com/microsoft/aspire/actions/runs/34268996559) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19985 |
| 2026-09-10 | [34432786645](https://github.com/microsoft/aspire/actions/runs/34432786645) | Tests / Package tests - macOS (class, Aspire.Templates.Tests, Templates-XUnit_V3MTP_NewUpAndBuildSupportP... / Templates-XUnit_V3MTP_NewUpAndBuildSupportProjectTemplatesTests (macos-latest) | #19999 |
| 2026-09-11 | [34586660260](https://github.com/microsoft/aspire/actions/runs/34586660260) | Tests / No-package tests (regular, Aspire.Cli.Tests, Cli, Cli, tests/Aspire.Cli.Tests/Aspire.Cli.Tests.cs... / Cli (ubuntu-latest) | #20005 |
| 2026-09-11 | [34655355873](https://github.com/microsoft/aspire/actions/runs/34655355873) | Tests / No-package tests (collection, Aspire.Hosting.Tests, Hosting-4, Hosting-4, tests/Aspire.Hosting.Tests) / Hosting-4 (windows-latest) | #20072 |
| 2026-09-12 | [34680944910](https://github.com/microsoft/aspire/actions/runs/34680944910) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20081 |
| 2026-09-14 | [34790579218](https://github.com/microsoft/aspire/actions/runs/34790579218) | Tests / No-package tests (regular, Aspire.Cli.Tests, Cli, Cli, tests/Aspire.Cli.Tests/Aspire.Cli.Tests.cs... / Cli (macos-latest) | #20079 |
| 2026-09-14 | [34795444609](https://github.com/microsoft/aspire/actions/runs/34795444609) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20062 |
| 2026-09-14 | [34803031152](https://github.com/microsoft/aspire/actions/runs/34803031152) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20092 |
| 2026-09-14 | [34806422379](https://github.com/microsoft/aspire/actions/runs/34806422379) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, winui-debug) | #19882 |
| 2026-09-14 | [34858055930](https://github.com/microsoft/aspire/actions/runs/34858055930) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19996 |
| 2026-09-14 | [34863279227](https://github.com/microsoft/aspire/actions/runs/34863279227) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19814 |
| 2026-09-15 | [35023188186](https://github.com/microsoft/aspire/actions/runs/35023188186) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20146 |
| 2026-09-15 | [35029892745](https://github.com/microsoft/aspire/actions/runs/35029892745) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20150 |
| 2026-09-16 | [35124724656](https://github.com/microsoft/aspire/actions/runs/35124724656) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20141 |
| 2026-09-17 | [35166494293](https://github.com/microsoft/aspire/actions/runs/35166494293) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20046 |
| 2026-09-18 | [35313069592](https://github.com/microsoft/aspire/actions/runs/35313069592) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20220 |
| 2026-09-18 | [35322237059](https://github.com/microsoft/aspire/actions/runs/35322237059) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #19887 |
| 2026-09-18 | [35337231287](https://github.com/microsoft/aspire/actions/runs/35337231287) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20230 |
| 2026-09-18 | [35377484081](https://github.com/microsoft/aspire/actions/runs/35377484081) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #20156 |
| 2026-09-21 | [35576439465](https://github.com/microsoft/aspire/actions/runs/35576439465) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Windows, dynamic-debug-configuration) | #1391 |

### LINKED ISSUE #19588: [CI Failure] Flaky: VS Code extension E2E (Linux, debug-dashboard) shard fails with generic exit code 1, unrelated to PR changes
## Build Information

Build: https://github.com/microsoft/aspire/actions/runs/32532172260
Build error leg: Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, debug-dashboard)
Pull request: #19552

## Error Message

```
Process completed with exit code 1. (Run extension E2E tests step failed; job logs unavailable: 'Failed to fetch logs for job'). No TRX test failures extracted. Occurs on 'VS Code extension E2E (Linux, debug-dashboard)' shard unrelated to PR changes touching only Aspire.Hosting executable launch argument computation.
```

## Description

Flaky: VS Code extension E2E (Linux, debug-dashboard) shard fails with generic exit code 1, unrelated to PR changes

**Type**: flaky-test

## Occurrences

| Date | Build | Job | PR |
|------|-------|-----|----|
| 2026-08-21 | [32532172260](https://github.com/microsoft/aspire/actions/runs/32532172260) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, debug-dashboard) | #19552 |
| 2026-08-22 | [32543765057](https://github.com/microsoft/aspire/actions/runs/32543765057) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, debug-dashboard) | #19131 |
| 2026-08-24 | [32755490521](https://github.com/microsoft/aspire/actions/runs/32755490521) | Tests / Hosting.Valkey / Hosting.Valkey (ubuntu-latest) | #19613 |

### LINKED ISSUE #19597: [CI Failure] Flaky: VS Code extension E2E (Linux, workspace-target-proof) shard fails with generic exit code 1, unrelated to PR changes
## Build Information

Build: https://github.com/microsoft/aspire/actions/runs/32550757022
Build error leg or test failing: Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask / `VS Code extension E2E (Linux, workspace-target-proof)`
Pull request: #19577

## Error Message

```
Process completed with exit code 1. (Run extension E2E tests step failed; job logs unavailable: 'Failed to fetch logs for job'). No TRX test failures extracted.
```

## Description

Flaky: VS Code extension E2E (Linux, workspace-target-proof) shard fails with generic exit code 1, unrelated to PR changes

**Type**: flaky-test

## Occurrences

| Date | Build | Job | PR |
|------|-------|-----|----|
| 2026-08-22 | [32550757022](https://github.com/microsoft/aspire/actions/runs/32550757022) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19577 |
| 2026-08-22 | [32551216411](https://github.com/microsoft/aspire/actions/runs/32551216411) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19494 |
| 2026-08-22 | [32554012820](https://github.com/microsoft/aspire/actions/runs/32554012820) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19516 |
| 2026-08-24 | [32675340121](https://github.com/microsoft/aspire/actions/runs/32675340121) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, workspace-target-proof) | #19044 |
| 2026-08-24 | [32689591099](https://github.com/microsoft/aspire/actions/runs/32689591099) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19429 |
| 2026-08-24 | [32755490521](https://github.com/microsoft/aspire/actions/runs/32755490521) | Tests / Hosting.Valkey / Hosting.Valkey (ubuntu-latest) | #19613 |
| 2026-08-24 | [32756925219](https://github.com/microsoft/aspire/actions/runs/32756925219) | Tests / Acquisition / Acquisition (windows-latest) | #19592 |
| 2026-08-24 | [32758526447](https://github.com/microsoft/aspire/actions/runs/32758526447) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19590 |
| 2026-08-24 | [32774802376](https://github.com/microsoft/aspire/actions/runs/32774802376) | Tests / Cli / Cli (windows-latest) | #19048 |
| 2026-08-24 | [32781735084](https://github.com/microsoft/aspire/actions/runs/32781735084) | Tests / Infrastructure / Infrastructure (ubuntu-latest) | #19624 |
| 2026-08-25 | [32785638179](https://github.com/microsoft/aspire/actions/runs/32785638179) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19583 |
| 2026-08-25 | [32814771945](https://github.com/microsoft/aspire/actions/runs/32814771945) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #18924 |
| 2026-08-25 | [32864200821](https://github.com/microsoft/aspire/actions/runs/32864200821) | Tests / Prepare Homebrew installer artifacts / Prepare Homebrew cask | #19627 |
| 2026-08-25 | [32880916910](https://github.com/microsoft/aspire/actions/runs/32880916910) | Tests / Dashboard / Dashboard (windows-latest) | #19620 |
| 2026-08-25 | [32885558008](https://github.com/microsoft/aspire/actions/runs/32885558008) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, workspace-target-proof) | #19615 |
| 2026-08-26 | [32994724407](https://github.com/microsoft/aspire/actions/runs/32994724407) | Tests / Run VS Code extension E2E tests / VS Code extension E2E (Linux, workspace-target-proof) | #19702 |

### PR DESCRIPTION
**Pull request by adamint**
## Description

Four VS Code extension E2E shards had recurring state/timing failures:

| Issue | Shard | Root cause | Fix |
| --- | --- | --- | --- |
| #19576 | Windows `dynamic-debug-configuration` | A cold single-file AppHost was still building when a fixed 60-second process-state wait expired; cleanup then hit `EBUSY`. | Give only that process-state wait 120 seconds and the test a matching extra 60 seconds of Mocha budget. |
| #19547 | Windows `edge-cases` | The debugger notification proved Python guidance fired, but the test used broad workbench text to prove a CodeLens rendered. | Query the actual CodeLens widgets with `waitForCodeLensText`. |
| #19597 | Linux `workspace-target-proof` | An unrelated `openAspireView()` could leave a blocking `ListError [aspire-vscode.appHosts] Invalid index 0` modal. The next hosted run exposed a second fixture boundary: a CLI wrapper stored inside a mutable workspace folder disappeared before reuse. | Remove only the incidental view action, keep repository/wizard/result proof, and place routing wrappers/logs in a stable sibling directory outside workspace folders. |
| #19588 | Linux `debug-dashboard` | PR #19131 fixed the original zero-second shutdown budget. A distinct recurring cascade remained: Linux js-debug can leave a late `Unable to launch browser` modal after the real `debugChrome` E2E finishes, blocking later tests indefinitely. | Keep deterministic unit coverage for the never-settling nonblocking launch contract; scope the real debugger-backed browser start/child-cleanup E2E to Windows, where the adapter is usable. |

All changes are test-only and state-specific. There are no new retries, global timeout changes, advisory shards, or production behavior changes.

Fixes #19576
Fixes #19547
Fixes #19597
Fixes #19588

### Coverage window

The renewed issue/run scan covers 2026-08-21T22:50Z through 2026-08-22T23:00Z:

- #19547: five matching Windows jobs (`96960020845`, `96971405255`, `96978370894`, `96981557279`, `96987091597`).
- #19597: five matching Linux collision-test jobs (`96971405336`, `96978370970`, `96981557316`, `96988100647`, `96987091575`).
- #19588: merged PR #19131 (`00bb3c9271a0d331ebafe7dad0cf0e927c6489ff`) owns the original zero-second shutdown fix. Runs 32543765057 and 32597141989 exposed the separate late-modal Selenium cascade addressed here.
- #19457's Java cleanup race was already fixed by PR #19474.
- #19453 remains a single pre-test restore timeout with no recurring signature.
- #19596 is CLI smoke/installer infrastructure, not a VS Code extension E2E shard.

The automated issue updater mislabeled several occurrence job names as Homebrew preparation jobs. The job IDs and raw logs above are authoritative. The final issue/health-ledger sweep found no new relevant VS Code E2E issue or occurrence after 2026-08-22T18:14:46Z.

### Validation

TDD source-contract proof covered each old failure mode before implementation:

```text
0 passing, 1 failing — expected edge-cases to use waitForCodeLensText(...)
0 passing, 1 failing — expected stable workspace-target CLI wrapper directory
0 passing, 1 failing — expected Windows-only real browser E2E plus deterministic nonblocking unit coverage
```

After the fixes:

```bash
corepack yarn compile-tests
corepack yarn unit-test --run out/test/e2eLaunchProfile.test.js --grep "uses state-specific assertions for debugger guidance and project creation readiness"
corepack yarn unit-test --run out/test/e2eLaunchProfile.test.js --grep "scopes real debug browser cleanup to Windows while retaining deterministic nonblocking coverage"
corepack yarn compile-e2e
corepack yarn lint
```

Focused contract tests passed; compile and lint succeeded. The browser contract normalizes CRLF before multiline matching, so Windows source checkouts are covered.

Focused macOS Extension Host runs:

- `edgeCases.e2e.test.js`: 6/6 passed.
- `workspaceTargetProof.e2e.test.js`: retained Linux modal signature absent; isolated routing path after wrapper isolation passed 1/1.
- `debugDashboard.e2e.test.js`: 7/7 passed with the platform-appropriate path; deterministic unit coverage retains the nonblocking never-settling launch regression.

Hosted evidence:

- Run 32595393972: Windows edge-cases, Windows dynamic-debug, Linux debug-dashboard, and the changed Linux collision test passed. It exposed the separate wrapper fixture failure fixed next.
- Run 32597141989 attempt 2: Linux workspace-target-proof, Windows edge-cases, and Windows dynamic-debug passed; Linux debug-dashboard reproduced the late js-debug modal cascade that led to the final Windows-only real-browser scope.
- Current-head run 32600262657 at `6767a9572a869a7a19ad6b8ce27347e2d3762fe7`: Windows extension unit tests, Windows dynamic-debug, Windows edge-cases, Linux workspace-target-proof, and Linux debug-dashboard all passed on the same head. The full run completed successfully on attempt 2.
- Windows ARM64 starter validation job 97098781009 was stuck in `Create starter app and validate startup` and GitHub cancelled it after 55 minutes without collecting diagnostics. A failed-job-only rerun (job 97105222886) passed in 10 minutes. This is outside the extension-only diff and matches the separately classified CLI starter infrastructure area.
- Other unrelated failures were classified, not folded into this PR: Windows CLI matches existing #19150; macOS starter validation is unrelated to this test-only diff.

This is a non-visual, test-only change. Screenshots, user-facing usage, security review, API review, and a breaking-change label do not apply.

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No

### CODE REVIEW

File: extension/src/test-e2e/dynamicDebugConfiguration.e2e.test.ts [hunk 1]
```diff
@@ -103,7 +107,9 @@ suite('Aspire dynamic debug configuration E2E', function () {
         const launch = await waitForLaunchOutput(beforeLaunch);
         assert.ok(launch.appHostPath);
         assert.ok(isSamePath(launch.appHostPath, ambiguousSecondAppHostPath));
-        const appHostPid = await waitForRunningAppHostPid(ambiguousSecondAppHostPath, 60000);
+        const appHostPid = await waitForRunningAppHostPid(
+            ambiguousSecondAppHostPath,
+            appHostProcessStateTimeoutMs);
```
[the remark this answers is not in the document]
Comment (adamint) [reply extension/src/test-e2e/dynamicDebugConfiguration.e2e.test.ts:113]: Fixed in `14a33a1d1`. This scenario now gets a 300-second Mocha budget, so the extra 60 seconds added to the process-state wait is usable without widening the timeout for the other tests. I also added a focused source-contract test that pins the 120-second wait and 300-second per-test deadline together.

**Review by ellahathaway — APPROVED** [review]
